hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/push/DelayedEventBusPushStrategy.java

```diff
@@ -163,38 +173,43 @@ private DispatchRunnable(final UI ui, final VaadinSession session) {
 
         @Override
         public void run() {
-            LOG.debug("UI EventBus aggregator started");
+            LOG.debug("UI EventBus aggregator started for UI {}", uiid);
             final long timestamp = System.currentTimeMillis();
-            final List<org.eclipse.hawkbit.eventbus.event.Event> events = new LinkedList<>();
-            for (int i = 0; i < BLOCK_SIZE; i++) {
+
+            final int size = queue.size();
+            if (size <= 0) {
+                LOG.debug("UI EventBus aggregator for UI {} has nothing to do.", uiid);
+                return;
+            }
+
+            final List<org.eclipse.hawkbit.eventbus.event.Event> events = new ArrayList<>(size);
+            for (int i = 0; i < size; i++) {
                 final org.eclipse.hawkbit.eventbus.event.Event pollEvent = queue.poll();
                 if (pollEvent == null) {
                     continue;
                 }
-                events.add(pollEvent);
+                events.add(i, pollEvent);
```

Comment:
There is no difference in using `events.add(pollEvent);` or using `events.add(i, pollEvent);` in this case, but when I'm thinking about it we actually changing the order of the events and maybe you were trying to fix this but then you would need to change for for-loop. The `queue.poll()` is given you a oldest event and this we add currently to the beginning of the list.

In my opinion the for-loop should like this
```
for(int i = size ; i >=0 ; i--)
{
   final org.eclipse.hawkbit.eventbus.event.Event pollEvent = queue.poll();
   if (pollEvent == null) {
      continue;
   }
   events.add(i, pollEvent);
}
```